tardis/montecarlo/src/integrator.c

```diff
@@ -258,11 +270,23 @@ _formal_integral(
                        ++patt_S_ul)
                     {
                       if (*pline < nu_end)
+                        // Calculate e-scattering optical depth to grid cell boundary
+                        zend = storage->time_explosion / C_INV * (1. - nu_end / nu);
+                        dtau += (zend - zstart) * escat_op;
                         break;
+
+                      // Calculate e-scattering optical depth to next resonance point
+                      zend = storage->time_explosion / C_INV * (1. - *pline / nu);
+                      dtau += (zend - zstart) * escat_op;
+
                       I_nu_r[p_idx] = I_nu_b[p_idx] * (*pexp_tau) + *patt_S_ul;
                       // In the absence of electron scattering, simple recursion as in Lucy 1991
                       // TODO: e-scattering: Replace by Lucy 1999, Eq. 27
                       I_nu_b[p_idx] = I_nu_r[p_idx];
+
+                      // reset e-scattering opacity 
+                      dtau = 0;
```

Comment:
Because dtau measures the accumulated e-scattering optical depth between consecutive resonances. I still have to modify the calculation of ``I_nu_b``, where this quantity is then used. Afterwards it has to be reset.